azurerm/internal/services/datalake/data_lake_store_virtual_network_rule_resource.go

```diff
@@ -0,0 +1,161 @@
+		Update: resourceDataLakeStoreVirtualNetworkRuleCreateUpdate,
+		Delete: resourceDataLakeStoreVirtualNetworkRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateDataLakeStoreVirtualNetworkRuleName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AccountName(),
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceDataLakeStoreVirtualNetworkRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Datalake.VirtualNetworkRulesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	virtualNetworkSubnetId := d.Get("subnet_id").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, accountName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Lake Store Virtual Network Rule %q (Account: %q, Resource Group: %q): %+v", name, accountName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_lake_store_virtual_network_rule", *existing.ID)
+		}
+	}
+
+	parameters := account.CreateOrUpdateVirtualNetworkRuleParameters{
+		CreateOrUpdateVirtualNetworkRuleProperties: &account.CreateOrUpdateVirtualNetworkRuleProperties{
+			SubnetID: utils.String(virtualNetworkSubnetId),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, accountName, name, parameters); err != nil {
+		return fmt.Errorf("Error creating Data Lake Store Virtual Network Rule %q (Account: %q, Resource Group: %q): %+v", name, accountName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Data Lake Store Virtual Network Rule %q (Account: %q, Resource Group: %q): %+v", name, accountName, resourceGroup, err)
+	}
+
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("Cannot read Data Lake Store Virtual Network Rule %q (Account: %q, Resource Group: %q)", name, accountName, resourceGroup)
+	}
+
+	d.SetId(*resp.ID)
```

Comment:
Since this is a new resource, you can drop the `Get` and use `d.SetId(id.ID())`

Reply:
Thank you for feedback. Makes sense. I committed proposed changes. PR is ready. Thanks!